iogt_content_migration/management/commands/load_v1_users.py

```diff
@@ -0,0 +1,365 @@
+
+            user_groups_cursor.close()
+
+    def migrate_user_groups(self):
+        self.stdout.write(self.style.SUCCESS('Starting User Groups Migration'))
+
+        sql = f'select * from auth_group'
+        cur = self.db_query(sql)
+
+        for row in cur:
+            group, __ = Group.objects.get_or_create(name=row['name'])
+            V1ToV2ObjectMap.create_map(group, row['id'])
+
+        self.stdout.write(self.style.SUCCESS('Completed User Groups Migration'))
+
+    def mark_user_registration_survey_required(self):
+        users = get_user_model().objects.filter(groups__id__in=self.registration_survey_mandatory_group_ids)
+        users.update(has_filled_registration_survey=False)
+
+    def migrate_user_comments(self):
+        self.stdout.write(self.style.SUCCESS('Starting Comment migration'))
+
+        sql = f'select * from django_comments dc inner join django_content_type dct on dc.content_type_id = dct.id'
+        cur = self.db_query(sql)
+
+        for row in self.with_progress(sql, cur, 'User comments migration in progress...'):
+            comment_id = row.pop('id')
+            content_type = self.content_type_map[row['model']]
+
+            if not content_type:
+                self.stdout.write(self.style.ERROR(f'Content Type for {row["model"]} not found.'))
+                continue
+
+            new_article = V1ToV2ObjectMap.get_v2_obj(Article, row['object_pk'])
+
+            max_thread_id_comment = XtdComment.objects.filter(
+                content_type_id=content_type.id, object_pk=new_article.pk).order_by('-thread_id').first()
+
+            max_thread_id = 0
+            if max_thread_id_comment:
+                max_thread_id = max_thread_id_comment.id
+
+            migrated_comment = V1ToV2ObjectMap.get_v2_obj(XtdComment, comment_id)
+
+            if not migrated_comment:
+                comment = XtdComment.objects.create(
+                    content_type_id=content_type.id, object_pk=new_article.pk, user_name=row['user_name'],
+                    user_email=row['user_email'], submit_date=row['submit_date'],
+                    comment=row['comment'], is_public=True, is_removed=False, thread_id=max_thread_id + 1,
+                    user=V1ToV2ObjectMap.get_v2_obj(get_user_model(), row['user_id']), order=1, followup=0,
+                    nested_count=0,
+                    site_id=1)
+                V1ToV2ObjectMap.create_map(comment, comment_id)
+
+    def migrate_user_survey_submissions(self):
+        sql = 'select * from surveys_molosurveysubmission mss ' \
+              'inner join surveys_molosurveypage msp on mss.page_id = msp.page_ptr_id'
+        cur = self.db_query(sql)
+
+        for row in self.with_progress(sql, cur, 'User Survey migration in progress...'):
+            try:
+                new_survey = V1ToV2ObjectMap.get_v2_obj(Survey, row['page_id'])
+                new_survey_page = Page.objects.get(pk=new_survey.id)
+            except KeyError:
```

Comment:
If new_survey is None (this happens in RKE because some surveys don't get saved due to too long Submit button text), we get an AttributeError rather than KeyError, and a crash. Catch that here as well.

Reply:
Didn't encounter this in the backup that I have. Will double check. Ideally, would want to throw this error to see what the issue is